extensions/raiderio/src/lookup-character.tsx

```diff
@@ -0,0 +1,55 @@
+import { Form, ActionPanel, Action, Cache, showToast, launchCommand, LaunchType } from "@raycast/api";
+import { REALMS } from "./constants";
+import { useState } from "react";
+
+const cache = new Cache();
+
+export default function Command() {
+  const [name, setName] = useState("");
+  const [realm, setRealm] = useState("");
+
+  function addToFavorites() {
+    const cachedFavorites = cache.get("favorites");
+    const favorites: Array<{ name: string; realm: string }> = cachedFavorites ? JSON.parse(cachedFavorites) : [];
+    if (
+      !favorites.some((favorite: { name: string; realm: string }) => favorite.name === name && favorite.realm === realm)
+    ) {
+      cache.set("favorites", JSON.stringify([...favorites, { name, realm }]));
+      showToast({
+        title: `Character ${name} on ${realm} added to favorites!`,
+      });
+      launchCommand({ name: "favorite-characters", type: LaunchType.UserInitiated });
+    } else {
+      showToast({
+        title: `Character ${name} is already a favorite`,
+      });
+    }
+  }
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.OpenInBrowser
+            title="Open RaiderIO Profile"
+            url={`https://raider.io/characters/us/${realm}/${name}`}
+          />
+          <Action title="Add to Favorites" onAction={addToFavorites} />
+          <Action
+            title="Open Favorites"
+            onAction={() => launchCommand({ name: "favorite-characters", type: LaunchType.UserInitiated })}
```

Comment:
We should maybe make this async and put into try{}catch{} to ensure it doesn't crash if the command is disabled? Why not let pinned items be on top of the list?